extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+using Microsoft.Azure.Functions.Worker.Core;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Newtonsoft.Json;
+using System.Linq;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Converter to bind Blob Storage type parameters.
+    /// </summary>
+    internal class BlobStorageConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context.Source is not IModelBindingData bindingData)
```

Comment:
As per my other comment, I don't see why we can't just bind to ModelBindingData here rather than an interface?